tools/python/transit/transit_graph_generator.py

```diff
@@ -0,0 +1,365 @@
+                id2 = stop2.get('transfer_id', stop2['id'])
+                seg = tuple(sorted([id1, id2]))
+                if seg in self.segments:
+                    edge['shape_ids'].append(self.segments[seg]['shape_id'])
+
+    def build(self):
+        if self.transit_graph is not None:
+            return self.transit_graph
+
+        self.__read_stops()
+        self.__read_transfers()
+        self.__read_networks()
+        self.__generate_transfer_nodes()
+        self.__generate_shapes()
+
+        self.transit_graph = {'networks': self.networks,
+                              'lines': self.lines,
+                              'gates': self.gates.values(),
+                              'stops': self.stops.values(),
+                              'transfers': self.transfers.values(),
+                              'shapes': self.shapes,
+                              'edges': self.edges}
+        return self.transit_graph
+
+    def show_preview(self):
+        for (s1, s2), info in self.segments.items():
+            plt.plot(info['curve'][:, 0], info['curve'][:, 1], 'g')
+        for stop in self.stops.values():
+            if 'transfer_id' in stop:
+                point = self.transfers[stop['transfer_id']]['point']
+                size = 60
+                color = 'r'
+            else:
+                point = stop['point']
+                if len(stop['line_ids']) > 2:
+                    size = 40
+                    color = 'b'
+                else:
+                    size = 20
+                    color = 'g'
+            plt.scatter([point['x']], [point['y']], size, color)
+        plt.show()
+
+
+def check_file_exists(file_path):
+    if not os.path.isfile(file_path):
+        print 'File', file_path, 'not found'
+        sys.exit(1)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input_file', help='input file name of transit data')
+    parser.add_argument('output_file', nargs='?', help='output file name of generated graph')
+    default_colors_path = os.path.dirname(os.path.abspath(__file__)) + '/../../../data/transit_colors.json'
+    parser.add_argument('-c', '--colors', type=str, default=default_colors_path,
+                        help='transit colors file COLORS_FILE_PATH', metavar='COLORS_FILE_PATH')
+    parser.add_argument('-p', '--preview', action="store_true", default=False,
+                        help="show preview of the transit scheme")
+
+    parser.add_argument('-a', '--alpha', type=float, default=0.5, help='the curves generator parameter value ALPHA',
+                        metavar='ALPHA')
+    parser.add_argument('-n', '--num', type=int, default=100, help='the number NUM of points in a generated curve',
+                        metavar='NUM')
+
+    args = parser.parse_args()
+
+    check_file_exists(args.input_file)
+    check_file_exists(args.colors)
+
+    with open(args.input_file, 'r') as input_file:
```

Comment:
Why do you need a check? You will have an io exception with a clear message here if file doesn't exist

Reply:
You're right, fixed.